docs/general/user-guide/0_getting-started.md

```diff
@@ -0,0 +1,181 @@
+# Getting Started 🚀
+
+## Main Concepts 🧠
+
+### DataFrame-Based Object-Oriented Framework 📊
+
+Unlike traditional mesa models where each agent is an individual Python object, mesa-frames stores all agents of a particular type in a single DataFrame. We operate only at the AgentSet level.
+
+This approach allows for:
+
+- Efficient memory usage
+- Improved performance through vectorized operations on agent attributes (This is what makes mesa-frames fast)
+
+Objects can be easily subclassed to respect mesa's object-oriented philosophy.
+
+### Vectorized Operations ⚡
+
+mesa-frames leverages the power of vectorized operations provided by DataFrame libraries:
+
+- Operations are performed on entire columns of data at once
+- This approach is significantly faster than iterating over individual agents
+- Complex behaviors can be expressed in fewer lines of code
+
+You should never use loops to iterate through your agents. Instead, use vectorized operations and implemented methods. If you need to loop, loop through vectorized operations (see the advanced tutorial SugarScape IG for more information).
+
+Check out these resources to understand vectorization and why it speeds up the code:
+
+- [What is vectorization?](https://stackoverflow.com/a/1422181)
+- [Vectorization Explained, Step by Step](https://machinelearningcompass.com/machine_learning_math/vectorization/)
+
+Here's a comparison between mesa-frames and mesa:
+
+=== "mesa-frames"
+    ```python
+    class MoneyAgentPolarsConcise(AgentSetPolars):
+        # initialization...
+
+        def give_money(self):
+            # Active agents are changed to wealthy agents
+            self.select(self.wealth > 0)
+
+            # Receiving agents are sampled (only native expressions currently supported)
+            other_agents = self.agents.sample(
+                n=len(self.active_agents), with_replacement=True
+            )
+
+            # Wealth of wealthy is decreased by 1
+            self["active", "wealth"] -= 1
+
+            # Compute the income of the other agents (only native expressions currently supported)
+            new_wealth = other_agents.group_by("unique_id").len()
+
+            # Add the income to the other agents
+            self[new_wealth, "wealth"] += new_wealth["len"]
+    ```
+
+=== "mesa"
+    ```python
+    class MoneyAgent(mesa.Agent):
+        # initialization...
+
+        def give_money(self):
+            # Verify agent has some wealth
+            if self.wealth > 0:
+                other_agent = self.random.choice(self.model.agents)
+                if other_agent is not None:
+                    other_agent.wealth += 1
+                    self.wealth -= 1
+    ```
+
+As you can see, while in mesa you should iterate through all the agents' steps in the model class, here you execute the method once for all agents.
```

Comment:
Needs to mention that in Mesa, https://github.com/projectmesa/mesa/blob/0b2d250cf2d3be79b266a04c0c91612880e28f0a/docs/tutorials/intro_tutorial.ipynb#L246,
> This can be important as the order in which agents are activated can impact the results of the model [Comer2014].
> [Comer2014] Comer, Kenneth W. “Who Goes First? An Examination of the Impact of Activation on Outcome Behavior in AgentBased Models.” George Mason University, 2014. http://mars.gmu.edu/bitstream/handle/1920/9070/Comer_gmu_0883E_10539.pdf

mesa-frames doesn't have this problem, but at the expense of having to carefully implement the conflict resolution, which you elaborated later in the tutorial.

Reply:
good point, added to the guide

Reply:
This paper was referenced by @jackiekazil during the meeting today. I'm highlighting this in your final GSoC evaluation that one of mesa-frames innovation is to constraint the user to design their model so that this problem is minimized.